keps/sig-api-machinery/3488-cel-admission-control/README.md

```diff
@@ -907,22 +917,26 @@ Problem examples:
 | CRD is in multiple clusters, but schema differs        | If policy author is aware of the schema variations, can they write policies that work for all the variations? |
 | Validation of an aggregated API server type            | Main API server does not have type definitions |
 
-Due to these complications, we have decided to evalute CEL expressions
-dynamically. Informational type checking will be provided (except for aggregated
-API server types), but will be surfaced only as warnings. See "Alternatives
-Considered" section for details of all the alternatives we reviewed when
-selecting this approach.
+Due to these complications, Type checking will be optional which can be activated by an expression-level
+`TypeCheckingMode` field. Possible modes are:
 
-Type checking is still performed for all expressions where a GVK can be matched
-to type check against, resulting in warnings, e.g.:
+- `Never` Do not type check the referred object or the param any time. 
```

Comment:
I could not see what `Background` means either. How do you think of `Unstructured`? This would expose the fact that we are treating `object`, `params`, etc as unstructured, which is strictly an implementation detail that worth hiding, but most precise in terms of what we want to express.

Reply:
The two type error options are:
- Report in status of the policy definition
- Type check errors result in admission failures and are reported in responses of admission requests

Maybe `TypeChecking: (Informative, RuntimeError)` ?

I donno. Maybe just leave this as-is and we can discuss in API review..

Reply:
Let's leave it as-is for now. `Never` is still valuable to silence all type-checking warning for these who know what they are doing.

Reply:
I've added an `Informative` that reflects the design before this PR>